app/controllers/users/contact_details.js

```diff
@@ -0,0 +1,174 @@
+import Ember from "ember";
+import AjaxPromise from "stock/utils/ajax-promise";
+import AsyncMixin, { ERROR_STRATEGIES } from "stock/mixins/async";
+import ImageUploadMixin from "stock/mixins/image_upload";
+
+export default Ember.Controller.extend(AsyncMixin, ImageUploadMixin, {
+  user: Ember.computed.alias("model.user"),
+
+  titles: Ember.computed(function() {
+    let translation = this.get("i18n");
+    let mr = translation.t("order.user_title.mr");
+    let mrs = translation.t("order.user_title.mrs");
+    let miss = translation.t("order.user_title.miss");
+    let ms = translation.t("order.user_title.ms");
+
+    return [
+      { name: mr, id: "Mr" },
+      { name: mrs, id: "Mrs" },
+      { name: miss, id: "Miss" },
+      { name: ms, id: "Ms" }
+    ];
+  }),
+
+  languages: Ember.computed(function() {
+    let translation = this.get("i18n");
+    let English = translation.t("users.languages.english");
+    let Chinese = translation.t("users.languages.chinese");
+
+    return [{ name: English, id: "en" }, { name: Chinese, id: "zh-tw" }];
+  }),
+
+  emailIdNotPresent: Ember.computed("user.email", function() {
+    let email = this.get("user.email");
+    return !/^[^@\s]+@[^@\s]+/.test(email);
+  }),
+
+  mobileNotPresent: Ember.computed("user.mobileWithoutCountryCode", function() {
+    let mobile = this.get("user.mobileWithoutCountryCode");
+    return !/^[456789]\d{7}/.test(mobile);
+  }),
+
+  districts: Ember.computed(function() {
+    return this.get("store")
+      .peekAll("district")
+      .sortBy("name");
+  }),
+
+  getUpdatedDistrictParams() {
+    let address_attributes = {
+      district_id: this.get("user.associatedDistrict.id") || null
+    };
+    return { address_attributes };
+  },
+
+  saveImage() {
+    let image = this.get("newUploadedImage");
+    if (image) {
+      return image.save();
+    }
+  },
+  deleteOldImage(img) {
+    if (img) {
+      img.deleteRecord();
```

Comment:
since this would fire an request, probably you can do some error handling as well